app/assets/javascripts/toggle_name_and_email_input.js

```diff
@@ -17,6 +17,10 @@
         {
           link.hide();
         }
+        else if ( targetScope === '[data-no-sunet-target]' )
+        {
+          $('form').find('input[type=text]').filter(':visible:first').focus();
```

Comment:
> On May 26, 2015, at 4:57 PM, Jessie Keck <notifications@github.com> wrote:
> 
> In app/assets/javascripts/toggle_name_and_email_input.js <https://github.com/sul-dlss/sul-requests/pull/154#discussion_r31092053>:
> 
> > @@ -17,6 +17,10 @@
> >          {
> >            link.hide();
> >          }
> > +        else if ( targetScope === '[data-no-sunet-target]' )
> > +        {
> > +          $('form').find('input[type=text]').filter(':visible:first').focus();
> I feel like this is scoped too broadly. If we were to add another form anywhere above this form in the DOM for any reason, this would unexpectedly attach focus to the first visible text input in that form.
> 
> Also, on line 10 there is a form variable (although I don't particularly like that either, because again, if we add a form after this one in the DOM we'll be scoped to the wrong form).
> 
Isn’t this form scoped by calling this function?

$(document).on('ready page:load', function(){
  $('[data-no-sunet="true"]').toggleNameAndEmailInputs();
});

So the last form would always be the sibling next to the toggled content?
> Here's a suggestion:
> 
> Change the variable in line 10 to use closest() <https://api.jquery.com/closest/> to find the parent form.
closest() broke it I think because it is looking up the DOM and the form is the last sibling.
> Change this line to be scoped to finding inputs within that form.
> —
> Reply to this email directly or view it on GitHub <https://github.com/sul-dlss/sul-requests/pull/154/files#r31092053>.
>